src/mqueryfront/src/QueryTimer.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from "react";
+
+class QueryTimer extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { currentTime: 0, countDownTime: undefined, firstCountDown: true };
+    }
+
+    tick() {
+        this.setState({
+            currentTime: Math.floor(Date.now() / 1000),
+        });
+    }
+
+    componentDidMount() {
+        this.interval = setInterval(() => this.tick(), 1000);
+
+        if (this.props.eta) {
+            this.interwalCountDown = setInterval(
```

Comment:
Typo?
```suggestion
            this.intervalCountDown = setInterval(
```
and this interval is never cancelled. Why do we need two intervals instead of one when they are ticking with the same frequency?